code/drasil-gool/GOOL/Drasil/LanguageRenderer.hs

```diff
@@ -702,62 +745,225 @@ classVarCheckStatic v = classVarCS (variableBind v)
           "classVar can only be used to access static variables"
         classVarCS Static = v
 
-classVarD :: (VariableSym repr) => repr (Type repr) -> 
-  repr (Variable repr) -> (Doc -> Doc -> Doc) -> repr (Variable repr)
-classVarD c v f = varFromData (variableBind v) 
-  (getTypeString c ++ "." ++ variableName v) 
-  (variableType v) (f (getTypeDoc c) (variableDoc v))
-
 classVarDocD :: Doc -> Doc -> Doc
 classVarDocD c v = c <> dot <> v
 
-objVarDocD :: VarData -> VarData ->  Doc
-objVarDocD n1 n2 = varDoc n1 <> dot <> varDoc n2
+objVarDocD :: Doc -> Doc ->  Doc
+objVarDocD n1 n2 = n1 <> dot <> n2
 
 inlineIfD :: ValData -> ValData -> ValData -> ValData
 inlineIfD c v1 v2 = vd prec (valType v1) (valDoc c <+> text "?" <+> 
   valDoc v1 <+> text ":" <+> valDoc v2)
   where prec = valPrec c <|> Just 0
 
-funcAppDocD :: Label -> [ValData] -> Doc
-funcAppDocD n vs = text n <> parens (valList vs)
-
-extFuncAppDocD :: Library -> Label -> [ValData] -> Doc
-extFuncAppDocD l n = funcAppDocD (l ++ "." ++ n)
+funcAppDocD :: (RenderSym repr) => Label -> [repr (Value repr)] -> Doc
+funcAppDocD n vs = text n <> parens (valueList vs)
 
-newObjDocD :: TypeData -> Doc -> Doc
-newObjDocD st vs = new <+> typeDoc st <> parens vs
+newObjDocD :: (RenderSym repr) => repr (Type repr) -> Doc -> Doc
+newObjDocD st vs = new <+> newObjDocD' st vs
 
-notNullDocD :: OpData -> ValData -> ValData -> Doc
-notNullDocD op v1 v2 = binOpDocD (opDoc op) (valDoc v1) (valDoc v2)
+newObjDocD' :: (RenderSym repr) => repr (Type repr) -> Doc -> Doc
+newObjDocD' st vs = getTypeDoc st <> parens vs
 
 listIndexExistsDocD :: OpData -> ValData -> ValData -> Doc
 listIndexExistsDocD greater lst index = parens (valDoc lst <> 
   text ".Length" <+> opDoc greater <+> valDoc index) 
 
+varD :: (RenderSym repr) => Label -> repr (Type repr) -> repr (Variable repr)
```

Comment:
It used to make sense to have all of these being named with a post-fix D, as they used to all operate on a `Doc`. Now that postfix D doesn't make as much sense...  I would seriously consider either dropping the D, or putting all of these in some kind of `.Generic` sub-module meant to be imported qualified. [But that can be done later]

Reply:
I intended the `D` here to stand for `Default` (whereas the functions for working with `Doc` end in `DocD`). I like the idea of putting them in a separate module to be imported qualified though. That would be less confusing.

Reply:
Indeed, I'd forgotten about `DocD`.  Such a move can be done incrementally, whenever you have to 'touch' some code anyways, then you can refactor those bits. Or if you're bored and looking for something straightforward to do 😄